### PR TITLE
chore: enable flake8 pre-commit check on e2e_tests

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -78,7 +78,8 @@ repos:
     rev: 3.9.2
     hooks:
       - id: flake8
-        exclude: ^(?!harness/).*|(harness/tests/experiment/fixtures/ancient-checkpoints/).*
+        files: ^(harness|e2e_tests)/.*
```
Comment (wes-turner) [inline .pre-commit-config.yaml:81]: nit: the trailing `.*` is extraneous without a terminating `$`. I think just `^(harness|e2e_tests)/` is a little clearer, here. Same is true with the exclude below.

But your choice. We're not at all consistent, stylewise, within this file.

**Review by wes-turner — APPROVED** [review]
This looks like a good change! Thanks!

Why, BTW, did you add this separate to #6695?

**Comment by hamidzr** [comment]
> Why, BTW, did you add this separate to https://github.com/determined-ai/determined/pull/6695?

just to have a smaller scope, each of these modules have a different flake8 config which don't help.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hamidzr) [reply .pre-commit-config.yaml:81]: true. I'll land it as is just to not sit through ci again but I'll update when I'm adding mypy support.

**Pull request merged**